eisbuk-admin/src/react-redux-firebase/reducer/index.ts

```diff
@@ -0,0 +1,93 @@
+import { Action } from "@/enums/store";
+
+import {
+  LocalStore,
+  FirestoreAction,
+  FirestoreReducerAction,
+  FirestoreState,
+} from "@/types/store";
+
+const defaultState: LocalStore["firestore"] = {
+  data: {},
+  listeners: {},
+};
+
+export default (
+  state: FirestoreState = defaultState,
+  action: FirestoreReducerAction<FirestoreAction>
+): FirestoreState => {
+  switch (action.type) {
+    case Action.UpdateFirestoreListener:
+      const { collection, listener } =
+        action.payload as FirestoreReducerAction<Action.UpdateFirestoreListener>["payload"];
+      return {
+        ...state,
+        listeners: {
+          ...state.listeners,
+          [collection]: { ...state.listeners[collection], ...listener },
+        },
+      };
+    case Action.DeleteFirestoreListener:
+      const collectionToDelete =
+        action.payload as FirestoreReducerAction<Action.DeleteFirestoreListener>["payload"];
+      // remove the data from the subscribed collection from the store
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { [collectionToDelete]: _coll, ...data } = state.data;
+      // remove the listener for a collection from the store
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { [collectionToDelete]: _listener, ...listeners } = state.listeners;
+      // return state with listener for collection and the correspoding data removed
+      return { data, listeners };
+
+    case Action.UpdateLocalCollection:
+      const {
+        collection: collectionToUpdate1,
+        data: updatedCollection1,
+        merge,
+      } = action.payload as FirestoreReducerAction<Action.UpdateLocalCollection>["payload"];
+      return {
+        data: {
+          ...state.data,
+          [collectionToUpdate1]: merge
+            ? { ...state.data[collectionToUpdate1], ...updatedCollection1 }
+            : updatedCollection1,
+        },
+        listeners: state.listeners,
+      };
+
+    case Action.UpdateLocalDocument:
+      const {
+        collection: collectionToUpdate2,
+        data: updatedDocument,
+        id: documentToUpdate,
+      } = action.payload as FirestoreReducerAction<Action.UpdateLocalDocument>["payload"];
+      return {
+        data: {
+          ...state.data,
+          [collectionToUpdate2]: {
```

Comment:
It looks like the `2` in the variable name is there to differentiate it from a different `collectionToUpdate` in the same block.

I propose to introduce blocks to scope variables when there's a similar need, instead of changing variable names. An opening bracket after the case statement (`case Action.UpdateLocalDocument: {`) and a closing one before the next `case` statement look more readable to me.

Reply:
BTW this should already be changed in the branch I'm working on. No action needed.